README.md

```diff
@@ -1,16 +1,225 @@
+| `npm run lint`             | Lints the project files                  |
+| `npm run lint:fix`         | Fixes lint errors in the project         |
+
+## Git flow
+
+### Pull requests
+
+- Use `git rebase` to merge brances locally.
+- Use `Squash and merge` option to merge PR into develop
+
+### Main branches
+
+- `main` - production branch
+- `develop` - development branch
+
+### Branch naming
+
+Use pattern: `<prefix>/<issue_id>/<short_description>`
+
+For example: `feature/41/navbar`
+
+Allowed prefixes are:
+
+- `feature` - for new feature
+- `hotfix` - for urgent fixes
+- `bugfix` - for fixing existing functionality
+
+## Project structure
+
+- **Project root**
+
+  - `/code` - root folder for application code
+    - `/config` - global config files (for tests, babel, webpack) and root html file
+    - `/reports` - reports regarding mutation testing
+    - `/src` - project source code folder
+      - `/assets` - for assets ('.png', '.svg', '.jpg', 'webp', ...)
+      - `/components` - for shared components
+      - `/constants` - for shared constants
+      - `/hooks` - for reusable hooks
+      - `/layouts` - for reusable relatively large page parts (headers, footers, sidebars, navbars, containers, etc.)
+      - `/pages` - for project pages
+      - `/store` - for globally managed application state
+      - `/styles` - for global styling and defining design system constants
+      - `/types` - for global types (utility types, business model types etc)
+      - `/utils` - for reusable functions
+    - `/stryker-config` - place for storing incremental file (for mutation tests)
+  - `/e2e` - for integration tests
+  - `/mock-backend` - express playground to test backend features while real backend is in process of developing
+
+## Styling
+
+### Motivation
+
+The general idea of our styling system is to create custom components (building blocks), that starts of prefix `App`, like `AppButton`, `AppTypography` etc. Then we use mui component inside as a base and add `scss` styles to it to customize it for our needs. Then we need to use `App`-prefixed component in our project and do not import it from mui directly. For building more complex components, like `containers`, `layouts` and `pages` we should use our wrappers around mui component. In that way we can easily migrate from mui to another ui package without huge refactorings.
+
+### Steps to create wrapper around mui component
+
+For example we want to create `Button` component. The following steps should be followed:
+
+1. Create folder, prefixed with `app`: `app-button`
+2. Create file prefixed with `App`: `AppButton.tsx`
+3. Create scss file for component: `AppButton.scss`
+4. Import scss file into `AppButton.tsx` using import alias: `import "@/components/app-button/AppButton.scss";`
+5. Import all scss variables from design system: `@import "@design-system";`
+6. Create base css class, prefixed with `spa`: `.spa-button`:
```

Comment:
Do we need this 'spa'?

Reply:
`spa` prefix is needed to create our custom design system with prefix of our company. Also it allows avoid collisions with other third-party css packages or another classes in general